packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -331,6 +334,27 @@ abstract class AnimatedWidgetBaseState<T extends ImplicitlyAnimatedWidget> exten
   void forEachTween(TweenVisitor<dynamic> visitor);
 }
 
+/// A base class for widgets with implicit animations that need to rebuild their
+/// Widget tree as the animation runs.
+///
+/// If you would rather respond to a running animation directly, use
+/// [ImplicitlyAnimatedWidgetState].
+///
+/// Subclasses must implement the [forEachTween] method to help
+/// [AnimatedWidgetBaseState] iterate through the subclasses' widget's fields
+/// and animate them.
+abstract class AnimatedWidgetBaseState<T extends ImplicitlyAnimatedWidget> extends ImplicitlyAnimatedWidgetState<T> {
+  @override
+  void initState() {
+    super.initState();
+    controller.addListener(_handleAnimationChanged);
+  }
+
+  void _handleAnimationChanged() {
+    setState(() { });
```

Comment:
add a comment inside the `{ }` explaining what state has changed and why we're calling setState. (Our style guide bans empty setStates, because it's usually a bad practice. This one must have snuck by at some point.)